### PR TITLE
PatchMode and PatchStatus parameters added to Set-AzVMOperatingSystem, and also 2020-12-01 functionality

### PR DESCRIPTION
**Pull request by Sandido**
Rest API change that introduced the breaking changes: https://github.com/Azure/azure-rest-api-specs/pull/12334
Notable changes:
1. Removed ContainerService cmdlets. This API was deprecated long ago. No customers use it successfully. https://azure.microsoft.com/en-us/updates/azure-container-service-will-retire-on-january-31-2020/ 
2. VMGuestPatch API updates included most of the breaking changes detected here: https://github.com/Azure/azure-rest-api-specs/pull/11768/files. These were approved by the Azure Breaking Change board. 

Azure/azure-powershell-cmdlet-review-pr#829
Azure/azure-powershell-cmdlet-review-pr#830

Swagger PR: Azure/azure-rest-api-specs#12334
.Net SDK PR: Azure/azure-sdk-for-net#18498

Work for:
https://msazure.visualstudio.com/One/_workitems/edit/9294531
https://msazure.visualstudio.com/One/_workitems/edit/9294548

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by Sandido** [comment]
Making ready for review so PS team can see the build errors related to breaking changes that were approved in Swagger and SDK with breaking change exceptions. Still some additional dev to be made.

**Comment by VeryEarly** [comment]
localfeed nupkg cannot be checked in.

Please also update *.test.csproj in modules referenced Compute, example:
https://github.com/Azure/azure-powershell/pull/14038/files

File: src/Compute/Compute/Compute.csproj
Comment (Sandido) [inline src/Compute/Compute/Compute.csproj:33]: Not sure why this is here.

Comment (VeryEarly) [reply src/Compute/Compute/Compute.csproj:33]: please remove if not needed

**Comment by VeryEarly** [comment]
> localfeed nupkg cannot be checked in.
> 
> Please also update *.test.csproj in modules referenced Compute, example:
> https://github.com/Azure/azure-powershell/pull/14038/files

Hi Adam, please fix these if you want this PR to catch this Azure-PowerShell release

**Review by VeryEarly — COMMENTED** [review]
looks like container service related changes can be removed from this PR

File: src/Compute/Compute/ChangeLog.md
Comment (VeryEarly) [inline src/Compute/Compute/ChangeLog.md:23]: I assume there is breaking change in this PR, can you please add a brief description:
"* [Breaking Change] ..."
in the Changelog.md

**Comment by Sandido** [comment]
We don't plan to release this publicly yet, rather in a preview module. Moving to this branch: https://github.com/Azure/azure-powershell/tree/ComputeIgnitePreview2

**Comment by Sandido** [comment]
moved to https://github.com/Azure/azure-powershell/pull/14315

**Comment by Sandido** [comment]
This work will be released in a preview module prior to 3/2. Moving to https://github.com/Azure/azure-powershell/pull/14319

**Comment by Sandido** [comment]
I just saw the email about the approved breaking changes. I will reopen and use this PR.

File: src/Compute/Compute/ChangeLog.md
Comment (Sandido) [reply src/Compute/Compute/ChangeLog.md:23]: @VeryEarly , the breaking change is the uptake of the new .Net SDK version 44. It changes the response to Get-AzVM for users whitelisted to the VMGuestPAtch API by removing and renaming 4 properties.

Comment (Sandido) [reply src/Compute/Compute/ChangeLog.md:23]: I'll add that.

**Comment by Sandido** [comment]
Hold off on merging for 1 more day please. We are trying to include these changes as well for the Ignite release: https://github.com/Azure/azure-powershell/pull/14325

**Comment by Sandido** [comment]
@VeryEarly , please proceed with merging this PR.

**Comment by grizzlytheodore** [comment]
please wait for https://github.com/Azure/azure-powershell/pull/14325 to merge in to this branch first before merging this PR. 

It has a CI failure that needs to be addressed by the PowerShell team. Please see: https://teams.microsoft.com/l/message/19:7599cc34651544b295371ded3fca0810@thread.skype/1614291370630?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=da8d67c5-f19d-4799-b158-5dbbef868d49&parentMessageId=1614291370630&teamName=Azure%20DevEx%20SDK%2C%20CLIs&channelName=Az%20PowerShell&createdTime=1614291370630

**Comment by grizzlytheodore** [comment]
@dcaro please see my comment above.  

I think the PowerShell team address it tonight, 
but if this PR needs to be merged tonight to release by 3/2, please go ahead

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**
